fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SubspaceProviderByKeySpacePath.java

```diff
@@ -38,26 +38,22 @@
     @Nonnull
     private final KeySpacePath keySpacePath;
 
-    @Nonnull
-    private final FDBRecordContext context;
-
     @Nullable
     private CompletableFuture<Subspace> subspaceFuture;
```

Comment:
If this provider is going to implement a cache, then it probably needs to be keyed by context.

It should be possible (if unusual) to take an open record store, turn it into a builder and open it with a context on another database (that is, with a different directory layer).

As mentioned in https://github.com/FoundationDB/fdb-record-layer/issues/338#issuecomment-457575052, though, I think it might be simpler to remember the `Subspace` inside the record store, which probably means that `KeySpacePath` can be a `SubspaceProvider` and there is no need for a separate `SubspaceProviderByKeySpacePath`.